### PR TITLE
Es6 lib/$

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- converts `lib/$` to es6
- adds a test for it
- rationalised `babelrc` a little

## What is the value of this and can you measure success?

- trying out conversion script
- getting jest running properly

## Tested in CODE?

- [x] tested

### CODE REVIEW

**Review by SiAdcock — APPROVED** [review]
Love it man. Them tests tho 👌 👌 👌

File: static/src/javascripts/lib/$.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    },
+});
+
+function $(selector, context) {
+    return bonzo(qwery(selector, context));
+}
+
+$.create = s => bonzo(bonzo.create(s));
+
+$.ancestor = (el, className) => {
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/$.js]: `Element.closest`? I don't think this is used anywhere though

Comment (gustavpursche) [inline static/src/javascripts/lib/$.js]: `lib/closest`, since `Element.closest` still has bad support :)

**Comment by sndrs** [comment]
fair few changes just gone in with typing stuff, would you mind looking over it again?

File: static/src/javascripts/lib/$.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if (el === null || el.nodeName.toLowerCase() === 'html') {
+        return false;
+    }
+    if (!el.parentNode || bonzo(el.parentNode).hasClass(className)) {
+        return el.parentNode;
+    }
+    return $.ancestor(el.parentNode, className);
+};
+
+$.forEachElement = (selector: string, fn: Function): Array<number> => {
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/$.js]: Isn't the return type `Array<Element>`?

**Comment by regiskuckaertz** [comment]
Nice 👍

File: static/src/javascripts/lib/$.spec.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    test(
+        'applies a function to an array of elements described by a selector',
+        () => {
+            $.forEachElement('.parent > *', el => el.classList.add('red'));
+            expect($('.child').hasClass('red')).toBe(true);
+            expect($('.sibling').hasClass('red')).toBe(true);
+        }
+    );
+
+    test("can find an element's ancestor", () => {
```
Comment (gtrufitt) [inline static/src/javascripts/lib/$.spec.js]: Why are these double quoteys here, thought we linted them?

Comment (sndrs) [inline static/src/javascripts/lib/$.spec.js]: `t's`, prettier converts them

File: static/src/javascripts/lib/$.js (hunk 2 above)
Comment (sndrs) [inline static/src/javascripts/lib/$.js]: er...yeah, looks like it! will check

**Pull request merged**